### PR TITLE
Add update and delete for volunteer

### LINKED ISSUE #31: Finish Up Volunteers CRUD Functionality
## Issue
- [ ] Style (non-breaking style code)
- [ ] Testing (add testing to code)
- [ ] Bug fix (change which fixes an issue)
- [ ] Refactor (improvement to the code)
- [x] New feature (change which adds functionality)
- [ ] Chore
## Issue Summary
:fire: add update action
:open_file_folder: files impacted: ``` /resources/volunteer.py ```  ``` /models/volunteer.py ```<br />
## Resources 
List any resources used when working on the issue (e.g. links to articles, stackoverflow, etc.)

### PR DESCRIPTION
**Pull request by adumortier**
## Who worked on this

- [x] Alexis
- [ ] Maria
- [ ] Ed
- [ ] Cody

## Type of change

- [ ] Style (Non-breaking Style code)
- [ ] Testing (Non-breaking Testing code)
- [ ] Bug fix or refactor (change which fixes an issue)
- [x] New feature (change which adds functionality)

## Summary of the change.

- Add `delete` and `update` methods for the `volunteers`
- Add conditionals when `volunteers` not found
- Add status code for each request on `volunteers`

🔥 If it is a new feature, describe the feature and the iteration of the project
spec it addresses.

Link to issue:

This closes #31 https://github.com/track-basket/trackbasket_BE/issues/31

## Checklist:

- [x] My code follows the Turing style guidelines
- [x] I have performed a self-review of my own code
- [x] I have commented my code if necessary

### CODE REVIEW

File: trackbasket_be/resources/volunteer.py [hunk 1]
```diff
@@ -6,11 +6,31 @@ class Volunteers(Resource):
 
   def get(self,id):
     volunteer = db.session.query(Volunteer).filter_by(volunteer_id=id).first()
-    return {'data': { 'id': 'volunteer', 'attributes': {'id': volunteer.volunteer_id, 'name': volunteer.name, 'phone number': volunteer.phone_number} } }
+    if volunteer is None:
```
Comment (adumortier) [inline trackbasket_be/resources/volunteer.py:9]: if no `volunteer` is found in the table the database query returns none

File: trackbasket_be/resources/volunteer.py [hunk 2]
```diff
@@ -6,11 +6,31 @@ class Volunteers(Resource):
...
+    else:  
+      return {'data': { 'id': 'volunteer', 'attributes': {'id': volunteer.volunteer_id, 'name': volunteer.name, 'phone number': volunteer.phone_number} } }, 200
 
   def post(self, id):
     request_data = request.json
     volunteer = Volunteer(name=request_data['name'], phone_number=request_data['phone_number'], volunteer_id=id)
     db.session.add(volunteer)
     db.session.commit()
-    return {'volunteer': request_data}
+    return {'volunteer': request_data}, 201
```
Comment (adumortier) [inline trackbasket_be/resources/volunteer.py:19]: added the status code for the successful post

File: trackbasket_be/resources/volunteer.py [hunk 3]
```diff
@@ -6,11 +6,31 @@ class Volunteers(Resource):
...
+  def patch(self, id):
+    request_data = request.json
+    volunteer = db.session.query(Volunteer).filter_by(volunteer_id=id).first()
+    if volunteer is None:
+      return {'data': { 'id': 'volunteer', 'attributes': {'error': "Volunteer not found"} } }, 400
+    else:
+      request_data = request.json
+      volunteer.name = request_data['name']
+      volunteer.phone_number = request_data['phone_number']
+      db.session.commit()
```
Comment (adumortier) [inline trackbasket_be/resources/volunteer.py:30]: we first retrieve the `volunteer` from the table, make changes to the `volunteer` instance and save it

File: trackbasket_be/resources/volunteer.py [hunk 4]
```diff
@@ -6,11 +6,31 @@ class Volunteers(Resource):
...
+      return {'data': { 'id': 'volunteer', 'attributes': {'id': volunteer.volunteer_id, 'name': volunteer.name, 'phone number': volunteer.phone_number} } }, 200
+
+  def delete(self, id):
+    db.session.query(Volunteer).filter_by(volunteer_id=id).delete()
+    db.session.commit()
+    return {'data': { 'id': 'volunteer', 'attributes': {'message': "volunteer with id {} successfully deleted".format(id)} } }, 200
```
Comment (adumortier) [inline trackbasket_be/resources/volunteer.py:36]: we will need to add another conditional here for when a `volunteer` was not found and could not be deleted

File: trackbasket_be/resources/volunteer.py (hunk 2 above)
Comment (mronauli) [reply trackbasket_be/resources/volunteer.py:19]: how about the case if a volunteer tries to create a new account even though they already have one?

Comment (mronauli) [reply trackbasket_be/resources/volunteer.py:19]: i like how you specify the status code

**Pull request merged**